src/error.rs

```diff
@@ -56,6 +63,63 @@ impl Error {
         self.inner.url.as_ref()
     }
 
+    /// Returns a mutable referene to the URL related to this error
+    ///
+    /// This is useful if you need to remove sensitive information from the URL
+    /// (e.g. an API key in the query), but do not want to remove the URL
+    /// entirely.
+    pub fn url_mut(&mut self) -> Option<&mut Url> {
+        self.inner.url.as_mut()
+    }
+
+    /// Set the URL related to this error
+    pub fn set_url(&mut self, url: Url) -> &mut Self {
+        self.inner.url = Some(url);
+        self
+    }
+
+    /// Clear a possible URL related to the error (in case it contains sensitive
+    /// information)
+    pub fn clear_url(&mut self) -> &mut Self {
+        self.inner.url = None;
+        self
+    }
+
+    /// Add a url related to this error (overwriting any existing)
+    pub fn with_url(mut self, url: Url) -> Self {
+        self.inner.url = Some(url);
+        self
+    }
+
+    /// Strip the related url from this error (if, for example, it contains
+    /// sensitive information)
+    pub fn without_url(mut self) -> Self {
+        self.inner.url = None;
+        self
+    }
+
+    /// Return a formatter for the error that masks any related URL
+    pub fn as_url_masked(&self) -> UrlMaskedError<'_> {
```

Comment:
What if we return an `impl Debug + Display`, instead of a named type, and in the debug format, don't show that it's wrapped.